bcipy/helpers/list.py

```diff
@@ -34,3 +34,47 @@ def grouper(iterable, chunk_size, incomplete="fill", fillvalue=None):
         return zip(*chunks)
 
     raise ValueError("Expected fill or ignore")
+
+
+def find_index(iterable: List,
+               match_item: Union[Any, Callable],
+               key: Callable = lambda x: x) -> Optional[int]:
+    """Find the index of the first item in the iterable which matches."""
+    for i, value in enumerate(iterable):
+        if callable(match_item):
+            result = match_item(value)
+        else:
+            result = key(value) == match_item
+        if result:
+            return i
+    return None
+
+
+def swapped(lst: List[Any], index1: int, index2: int) -> List[Any]:
+    """Creates a copy of the provided list with elements at the given indices
+    swapped."""
+    replacements = {index1: lst[index2], index2: lst[index1]}
+    return [replacements.get(i, val) for i, val in enumerate(lst)]
+
+
+def expanded(lst: List[Any],
+             length: int,
+             fill: Union[Any, Callable] = lambda x: x[-1]) -> List[Any]:
+    """Creates a copy of the provided list expanded to the given length. By
+    default the last item is used as the fill item.
+
+    Parameters
+    ----------
+        lst - list of items to copy
+        length - expands list to this length
+        fill - optional; used to determine which element to use for
+            the fill, given the list. Defaults to the last element.
+
+    >>> expand([1,2,3], length=5)
+    [1,2,3,3,3]
+    """
+    times = length - len(lst)
+    if lst and times > 0:
```

Comment:
Given the type, we can assume this is not None. Perhaps a lst length check?

Reply:
This is checking for an empty list. See https://peps.python.org/pep-0008/.